hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/filter/PathFilter.java

```diff
@@ -1,58 +1,128 @@
+    private static final String ARTHAS_START = "arthas";
+
     public static final String REQUEST_TIME = "request_time";
     public static final String REQUEST_PARAMS_JSON = "request_params_json";
 
+    private static final String DELIMETER = "/";
+    private static final Set<String> WHITE_API_LIST = ImmutableSet.of(
+            "",
+            "apis",
+            "metrics",
+            "versions",
+            "health",
+            "gremlin",
+            "graphs/auth",
+            "graphs/auth/users",
+            "auth/users",
+            "auth/managers",
+            "auth",
+            "hstore",
+            "pd",
+            "kafka",
+            "whiteiplist",
+            "vermeer",
+            "store",
+            "expiredclear",
+            "department",
+            "saas",
+            "trade",
+            "kvstore",
+            "openapi.json"
+    );
+
+    @Context
+    private jakarta.inject.Provider<HugeConfig> configProvider;
+
+    public static boolean isWhiteAPI(String rootPath) {
+
+        return WHITE_API_LIST.contains(rootPath);
+    }
+
     @Override
-    public void filter(ContainerRequestContext context) throws IOException {
+    public void filter(ContainerRequestContext context)
+            throws IOException {
         context.setProperty(REQUEST_TIME, System.currentTimeMillis());
 
-        // TODO: temporarily comment it to fix loader bug, handle it later
-        /*// record the request json
-        String method = context.getMethod();
-        String requestParamsJson = "";
-        if (method.equals(HttpMethod.POST)) {
-            requestParamsJson = IOUtils.toString(context.getEntityStream(),
-                                                 Charsets.toCharset(CHARSET));
-            // replace input stream because we have already read it
-            InputStream in = IOUtils.toInputStream(requestParamsJson, Charsets.toCharset(CHARSET));
-            context.setEntityStream(in);
-        } else if (method.equals(HttpMethod.GET)) {
-            MultivaluedMap<String, String> pathParameters = context.getUriInfo()
-                                                                   .getPathParameters();
-            requestParamsJson = pathParameters.toString();
+        List<PathSegment> segments = context.getUriInfo().getPathSegments();
+        E.checkArgument(segments.size() > 0, "Invalid request uri '%s'",
+                        context.getUriInfo().getPath());
+        String rootPath = segments.get(0).getPath();
+
+        if (isWhiteAPI(rootPath)) {
+            return;
+        }
+
+        if (GRAPH_SPACE.equals(rootPath)) {
+            return;
+        }
+
+        if (ARTHAS_START.equals(rootPath)) {
+            return;
         }
 
-        context.setProperty(REQUEST_PARAMS_JSON, requestParamsJson);*/
+        UriInfo uriInfo = context.getUriInfo();
+        String defaultPathSpace =
+                this.configProvider.get().get(ServerOptions.PATH_GRAPH_SPACE);
```

Comment:
‼️ **URI 拼接逻辑存在严重问题**

当前的路径拼接方式会导致 URI 格式错误：

1. `uriInfo.getBaseUri().getPath()` 已经包含末尾的 `/`
2. 使用 `String.join(DELIMETER, ...)` 会在每个部分之间都添加 `/`
3. 最终会产生多余的斜杠，例如：`/api//graphspaces//default//graphs`

**建议修改：**
```suggestion
StringBuilder pathBuilder = new StringBuilder();
pathBuilder.append(uriInfo.getBaseUri().getPath());
if (!pathBuilder.toString().endsWith(DELIMETER)) {
    pathBuilder.append(DELIMETER);
}
pathBuilder.append(GRAPH_SPACE).append(DELIMETER).append(defaultPathSpace);
for (PathSegment segment : segments) {
    pathBuilder.append(DELIMETER).append(segment.getPath());
}
String path = pathBuilder.toString();
```